Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/conceptcollection/ConceptCollectionManagerTest.java

```diff
@@ -0,0 +1,209 @@
+
+	private Concept concept;
+
+	@Before
+	public void setUp() throws QuadrigaStorageException {
+		MockitoAnnotations.initMocks(this);
+
+		List<ConceptpowerReply.ConceptEntry> conceptEntries = new ArrayList<ConceptpowerReply.ConceptEntry>();
+		ConceptpowerReply.ConceptEntry entry = new ConceptpowerReply.ConceptEntry();
+		entry.setLemma("test-lemma");
+		conceptEntries.add(entry);
+		ConceptpowerReply rep = new ConceptpowerReply();
+		rep.setConceptEntry(conceptEntries);
+
+		Mockito.when(mockedRestTemplate.getForObject(Matchers.anyString(), Matchers.eq(ConceptpowerReply.class),
+				Matchers.anyMapOf(String.class, String.class))).thenReturn(rep);
+
+		Project project = new Project();
+		project.setCreatedBy("createdBy");
+		project.setProjectName("test project");
+		project.setProjectId("id");
+
+		Project project2 = new Project();
+		project2.setCreatedBy("createdBy2");
+		project2.setProjectName("test project2");
+		project2.setProjectId("id2");
+
+		List<IProject> ccProjectsList = new ArrayList<IProject>();
+		ccProjectsList.add(project);
+
+		List<IProject> ccProjectsList2 = new ArrayList<IProject>();
+		ccProjectsList2.add(project);
+		ccProjectsList2.add(project2);
+
+		Mockito.when(mockedProjectShallowMapper.getProjectList(Matchers.anyString())).thenReturn(ccProjectsList2);
+		Mockito.when(mockedProjectShallowMapper.getCollaboratorProjectListOfUser(Matchers.anyString()))
+				.thenReturn(ccProjectsList);
+
+		WorkSpace workspace = new WorkSpace();
+		workspace.setWorkspaceId("w-id");
+		workspace.setWorkspaceName("w-name");
+
+		WorkSpace workspace2 = new WorkSpace();
+		workspace2.setWorkspaceId("w-id2");
+		workspace2.setWorkspaceName("w-name2");
+
+		List<IWorkSpace> ccWorkspaceList = new ArrayList<IWorkSpace>();
+		ccWorkspaceList.add(workspace);
+
+		List<IWorkSpace> ccWorkspaceList2 = new ArrayList<IWorkSpace>();
+		ccWorkspaceList2.add(workspace);
+		ccWorkspaceList2.add(workspace2);
+		Mockito.when(mockedwsListManger.getWorkspaceByConceptCollection(Matchers.anyString()))
+				.thenReturn(ccWorkspaceList);
+		Mockito.when(mockedwsManager.listActiveWorkspace(Matchers.anyString(), Matchers.anyString()))
+				.thenReturn(ccWorkspaceList2);
+
+		concept = new Concept();
+
+		Mockito.when(mockedConceptFactory.createConceptObject()).thenReturn(concept);
+		Mockito.when(mockedccDao.updateItem(Matchers.any(IConcept.class), Mockito.anyString(), Mockito.anyString()))
+				.thenReturn("dummy");
+	}
+
+	@Test
+	public void testSearch() {
+
+		ConceptpowerReply reply = conceptCollectionManagerUnderTest.search("item", "pos");
+		assertNotNull(reply);
+	}
+
+	@Test
+	public void testSearchNegative() {
+
+		ConceptpowerReply reply = conceptCollectionManagerUnderTest.search(null, null);
+		assertNull(reply);
+	}
+
+	@Test
+	public void getConceptLemmaFromConceptIdTest() {
+
+		String result = conceptCollectionManagerUnderTest.getConceptLemmaFromConceptId("id");
+		assertEquals("test-lemma", result);

```

Comment:
That's not reallly testing if the method works correct. You should at least make sure the rest template is called with the correct url, otherwise the method could call the rest template with an empty string and your test would still pass.